general/array.hpp

```diff
@@ -92,6 +93,10 @@ class Array
    template <typename CT, int N>
    explicit inline Array(const CT (&values)[N]);
 
+   /// Initilizer list contructor for syntax like Array<int> a = {1,2,3,4,5};
+   inline Array(std::initializer_list<T> list) 
```

Comment:
I've never used one of these before so someone should make sure I didn't munge something that my quick test didn't pick up.